bcipy/task/paradigm/rsvp/copy_phrase.py

```diff
@@ -100,8 +101,8 @@ def __init__(self, win, daq, parameters, file_save, signal_model,
         self.window = win
         self.daq = daq
         self.parameters = parameters
-        for param in RSVPCopyPhraseTask.PARAMETERS_USED:
-            assert param in self.parameters, f"parameter '{param}' is required"
+
+        self.validate()
```

Comment:
I like this pattern! We should consider adding it to the abstract parent class as a reminder when we are implementing other tasks that there may be some assumptions about the values and their relationships.

We may also want to encode constraints directly in the parameters.json file (ex. min/max values, etc) and have these be validated by the Parameters module, but there should still be some validations that are task-specific.

What do you think about making this more specific to parameters: `self.validate_parameters()` or `self.validate_parameters(parameters)`?

Reply:
I like that suggestion and added it to the PR. There are a few improvements we should address to parameter setting in general! I've made a ticket for now that we'll need to breakdown during our next refinement. https://www.pivotaltracker.com/story/show/181940742